skrf/calibration/deembedding.py

```diff
@@ -0,0 +1,452 @@
+'''
+.. module:: skrf.calibration.deembedding
+
+====================================================
+deembedding (:mod:`skrf.calibration.deembedding`)
+====================================================
+
+This module provides objects to implement de-embedding methods
+that are often used for on-wafer applications. 
+Each de-embedding method inherits from the common abstract base 
+class :class:`Deembedding`.
+
+Base Class
+---------------
+
+.. autosummary::
+   :toctree: generated/
+
+   Deembedding
+
+De-embedding Methods
+-----------------------
+
+.. autosummary::
+   :toctree: generated/
+
+   OpenShort
+   Open
+   ShortOpen
+   Short
+'''
+
+from abc import ABC, abstractmethod
+from ..frequency import *
+from ..network import *
+
+class Deembedding(ABC):
+    '''
+    Abstract Base Class for all de-embedding objects.
+
+    This class implements the common mechanisms for all de-embedding
+    algorithms. Specific calibration algorithms should inherit this
+    class and over-ride the methods:
+    * :func:`Deembedding.deembed`
+
+    '''
+    def __init__(self, dummies, name=None, *args, **kwargs):
+        '''
+        De-embedding Initializer
+
+        Notes
+        -----
+        Each de-embedding algorithm may use a different number of
+        dummy networks. We check that each of these dummy networks
+        have matching frequecies to perform de-embedding.
+
+        It should be known a-priori what the equivalent circuit 
+        of the parasitic network looks like. The proper de-embedding
+        method should then be chosen accordingly.
+
+        Parameters
+        ----------
+        dummies : list of :class:`~skrf.network.Network` objects
+            Network info of all the dummy structures used in a 
+            given de-embedding algorithm.
+
+        name : string
+            Name of this de-embedding instance, like 'open-short-set1'
+            This is for convenience of identification.
+
+        \*args, \*\*kwargs : keyword arguments
+            stored in self.args and self.kwargs, which may be used
+            by sub-classes if needed.
+        '''
+
+       # ensure all the dummy Networks' frequency's are the same
+        for dmyntwk in dummies:
+            if dummies[0].frequency != dmyntwk.frequency:
+                raise(ValueError('Dummy Networks dont have matching frequencies.')) 
+
+        # TODO: attempt to interpolate if frequencies do not match
+
+        self.args = args
+        self.kwargs = kwargs
+        self.name = name
+
+    def __str__():
+        pass
+
+    def __repr_():
+        pass

```

Comment:
These two methods are required to have 'self' as first argument. str(instance) throws an error now.
However, as they are empty anyway I'd suggest to remove them completely.
If you want Child classes to override these methods, you should make them abstract.

Reply:
Thanks for pointing that out. I forgot to update those methods, I'll push an update shortly.